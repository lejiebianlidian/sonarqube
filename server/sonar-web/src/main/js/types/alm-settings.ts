/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
export const enum AlmKeys {
  Azure = 'azure',
  Bitbucket = 'bitbucket',
  GitHub = 'github',
  GitLab = 'gitlab'
}

export interface AlmBindingDefinition {
  key: string;
}

export interface AzureBindingDefinition extends AlmBindingDefinition {
  personalAccessToken: string;
}

export interface BitbucketBindingDefinition extends AlmBindingDefinition {
  personalAccessToken: string;
  url: string;
}

export interface GithubBindingDefinition extends AlmBindingDefinition {
  appId: string;
  clientId: string;
  clientSecret: string;
  privateKey: string;
  url: string;
}

export interface GitlabBindingDefinition extends AlmBindingDefinition {
  personalAccessToken: string;
  url?: string;
}

export interface ProjectAlmBindingResponse {
  alm: AlmKeys;
  key: string;
  repository?: string;
  slug?: string;
  summaryCommentEnabled?: boolean;
}

export interface ProjectBitbucketBindingResponse extends ProjectAlmBindingResponse {
  alm: AlmKeys.Bitbucket;
  repository: string;
  slug: string;
}

export interface ProjectAlmBindingParams {
  almSetting: string;
  project: string;
}

export interface AzureProjectAlmBindingParams extends ProjectAlmBindingParams {}

export interface BitbucketProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository: string;
  slug: string;
}

export interface GithubProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository: string;
  summaryCommentEnabled: boolean;
}

export interface GitlabProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository?: string;
}

export interface AlmSettingsInstance {
  alm: AlmKeys;
  key: string;
  url?: string;
}

export interface AlmSettingsBindingDefinitions {
  [AlmKeys.Azure]: AzureBindingDefinition[];
  [AlmKeys.Bitbucket]: BitbucketBindingDefinition[];
  [AlmKeys.GitHub]: GithubBindingDefinition[];
  [AlmKeys.GitLab]: GitlabBindingDefinition[];
}
