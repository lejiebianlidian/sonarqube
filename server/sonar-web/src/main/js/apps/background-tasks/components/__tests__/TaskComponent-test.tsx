/*
 * SonarQube
 * Copyright (C) 2009-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { ComponentQualifier } from '../../../../types/component';
import TaskComponent from '../TaskComponent';

it('renders correctly', () => {
  expect(shallowRender()).toMatchSnapshot();
  expect(shallowRender({ componentKey: undefined })).toMatchSnapshot('undefined key');
  expect(shallowRender({ componentQualifier: ComponentQualifier.Portfolio })).toMatchSnapshot(
    'portfolio'
  );
  expect(shallowRender({ branch: 'feature' })).toMatchSnapshot('branch');
  expect(shallowRender({ branch: 'branch-6.7' })).toMatchSnapshot('branch');
  expect(shallowRender({ pullRequest: 'pr-89' })).toMatchSnapshot('pull request');
});

function shallowRender(taskOverrides: Partial<T.Task> = {}) {
  const TASK = {
    componentKey: 'foo',
    componentName: 'foo',
    componentQualifier: 'TRK',
    id: 'bar',
    organization: 'org',
    status: 'PENDING',
    submittedAt: '2017-01-01',
    submitterLogin: 'yoda',
    type: 'REPORT'
  };
  return shallow(<TaskComponent task={{ ...TASK, ...taskOverrides }} />);
}
