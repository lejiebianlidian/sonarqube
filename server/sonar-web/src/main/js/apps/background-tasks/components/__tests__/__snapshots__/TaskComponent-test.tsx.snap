// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<td>
  <span
    className="little-spacer-right"
  >
    <QualifierIcon
      qualifier="TRK"
    />
  </span>
  <Connect(Organization)
    organizationKey="org"
  />
  <Link
    className="spacer-right"
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/dashboard",
        "query": Object {
          "branch": undefined,
          "id": "foo",
        },
      }
    }
  >
    foo
  </Link>
  <TaskType
    type="REPORT"
  />
</td>
`;

exports[`renders correctly: branch 1`] = `
<td>
  <BranchIcon
    className="little-spacer-right"
  />
  <Connect(Organization)
    organizationKey="org"
  />
  <Link
    className="spacer-right"
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/dashboard",
        "query": Object {
          "branch": "feature",
          "id": "foo",
        },
      }
    }
  >
    foo
    <span
      className="text-limited text-text-top"
      title="feature"
    >
      <span
        style={
          Object {
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      >
        /
      </span>
      feature
    </span>
  </Link>
  <TaskType
    type="REPORT"
  />
</td>
`;

exports[`renders correctly: branch 2`] = `
<td>
  <BranchIcon
    className="little-spacer-right"
  />
  <Connect(Organization)
    organizationKey="org"
  />
  <Link
    className="spacer-right"
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/dashboard",
        "query": Object {
          "branch": "branch-6.7",
          "id": "foo",
        },
      }
    }
  >
    foo
    <span
      className="text-limited text-text-top"
      title="branch-6.7"
    >
      <span
        style={
          Object {
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      >
        /
      </span>
      branch-6.7
    </span>
  </Link>
  <TaskType
    type="REPORT"
  />
</td>
`;

exports[`renders correctly: portfolio 1`] = `
<td>
  <span
    className="little-spacer-right"
  >
    <QualifierIcon
      qualifier="VW"
    />
  </span>
  <Connect(Organization)
    organizationKey="org"
  />
  <Link
    className="spacer-right"
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/portfolio",
        "query": Object {
          "id": "foo",
        },
      }
    }
  >
    foo
  </Link>
  <TaskType
    type="REPORT"
  />
</td>
`;

exports[`renders correctly: pull request 1`] = `
<td>
  <PullRequestIcon
    className="little-spacer-right"
  />
  <Connect(Organization)
    organizationKey="org"
  />
  <Link
    className="spacer-right"
    onlyActiveOnIndex={false}
    style={Object {}}
    to={
      Object {
        "pathname": "/dashboard",
        "query": Object {
          "id": "foo",
          "pullRequest": "pr-89",
        },
      }
    }
  >
    foo
    <span
      className="text-limited text-text-top"
    >
      <span
        style={
          Object {
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      >
        /
      </span>
      pr-89
    </span>
  </Link>
  <TaskType
    type="REPORT"
  />
</td>
`;

exports[`renders correctly: undefined key 1`] = `
<td>
  <span
    className="note"
  >
    bar
  </span>
  <TaskType
    type="REPORT"
  />
</td>
`;
